From 2d2fba7a2f9ba37bcce29269618e12a336766cc6 Mon Sep 17 00:00:00 2001
From: Arie Brown <abrown@habana.ai>
Date: Thu, 28 Apr 2022 22:53:35 +0300
Subject: [PATCH] [SW-72831] gaudi2: uart: fix mask of device id

using incorrect mask when reading device id cause showing 0 as device
id.

Change-Id: I3c9d56f968b1c9458d40773f72e12a47da8d6e86
Signed-off-by: Arie Brown <abrown@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 9f42a2c..1bd6eda 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -371,7 +371,7 @@
 }
 
 #define HL_GAUDI2_MEZZ_ID_SHIFT	2
-#define HL_GAUDI2_MEZZ_ID_MASK	0x1C /* GENMASK(4, 2), use 3 out of 5 */
+#define HL_GAUDI2_MEZZ_ID_MASK	0x7 /* We use only 3 bits out of 5 */
 
 void hl_esc_key_handle(void)
 {
